perfkitbenchmarker/benchmarks/mysql_service_benchmark.py

```diff
@@ -0,0 +1,838 @@
+    results: The dictionary to store results based on sysbench output.
+    metadata: The metadata to be passed along to the Samples class.
+  """
+  all_tps = []
+  seen_general_statistics = False
+  seen_response_time = False
+
+  response_times = {}
+
+  sysbench_output_io = StringIO.StringIO(sysbench_output)
+  for line in sysbench_output_io.readlines():
+    if re.match('^\[', line):
+      tps = re.findall('tps: (.*?),', line)
+      all_tps.append(float(tps[0]))
+      continue
+
+    if line.startswith('General statistics:'):
+      seen_general_statistics = True
+      continue
+
+    if seen_general_statistics:
+      if re.match('^ +response time:.*', line):
+        seen_response_time = True
+        continue
+
+    if seen_general_statistics and seen_response_time:
+      for token in RESPONSE_TIME_TOKENS:
+        search_string = '.*%s: +(.*)ms' % token
+        if re.findall(search_string, line):
+          response_times[token] = float(re.findall(search_string, line)[0])
+
+  tps_line = ', '.join(map(str, all_tps))
+  # Print all tps data points in the log for reference. And report
+  # percentiles of these tps data in the final result set.
+  logging.info('All TPS numbers: \n %s', tps_line)
+
+  tps_percentile = _PercentileCalculator(all_tps)
+  for percentile in PERCENTILES_LIST:
+    percentile_string = 'p%s' % str(percentile)
+    logging.info('%s tps %f', percentile_string,
+                 tps_percentile[percentile_string])
+    metric_name = ('%s %s') % (SYSBENCH_RESULT_NAME_TPS, percentile_string)
+    results.append(sample.Sample(
+        metric_name,
+        tps_percentile[percentile_string],
+        NA_UNIT,
+        metadata))
+
+  # Also report average, stddev, and coefficient of variation
+  for token in ['average', 'stddev']:
+    logging.info('tps %s %f', token, tps_percentile[token])
+    metric_name = ('%s %s') % (SYSBENCH_RESULT_NAME_TPS, token)
+    results.append(sample.Sample(
+        metric_name,
+        tps_percentile[token],
+        NA_UNIT,
+        metadata))
+
+  if tps_percentile['average'] > 0:
+    cv = tps_percentile['stddev'] / tps_percentile['average']
+    logging.info('tps coefficient of variation %f', cv)
+    metric_name = ('%s %s') % (SYSBENCH_RESULT_NAME_TPS, 'cv')
+    results.append(sample.Sample(
+        metric_name,
+        cv,
+        NA_UNIT,
+        metadata))
+
+  # Now, report the latency numbers.
+  for token in RESPONSE_TIME_TOKENS:
+    logging.info('%s_response_time is %f', token, response_times[token])
+    metric_name = '%s %s' % (SYSBENCH_RESULT_NAME_LATENCY, token)
+
+    if token == 'percentile':
+      metric_name = '%s %s' % (metric_name, FLAGS.sysbench_latency_percentile)
+
+    results.append(sample.Sample(
+        metric_name,
+        response_times[token],
+        MS_UNIT,
+        metadata))
+
+
+def _IssueSysbenchCommand(vm, duration):
+  """ Issues a sysbench run command given a vm and a duration. Does nothing
```

Comment:
Docstrings should contain a summary line followed by details - see [here](https://google-styleguide.googlecode.com/svn/trunk/pyguide.html?showone=Comments#Comments).